hail/python/hailtop/hailctl/auth/login.py

```diff
@@ -0,0 +1,99 @@
+import os
+import socket
+import asyncio
+import webbrowser
+import aiohttp
+from aiohttp import web
+
+from hailtop.gear import get_deploy_config
+from hailtop.gear.auth import get_tokens, auth_headers
+
+
+def init_parser(parser):  # pylint: disable=unused-argument
```

Comment:
Could we add tests for login, logout?

Reply:
I don't actually know how to do that, it would involve automating the Google oauth2 flow.  I'm not actually sure how to handle it.  I'm sure it's possible, so one option is to automate.  One option is to make this a manual test that someone has to run in a test namespace and attest to it.  Another option is a staging-like step as part of deploy where things like this and testing the UI more generally gate production deployment.  I think the second option is good for now, and we can continue to think about it.

Reply:
Sure, #2 sounds reasonable. I think the typical way to do this would be to mock the service. Google uses this approach: https://github.com/googleapis/google-auth-library-python-oauthlib/blob/master/tests/test_flow.py


```python
authorization_url_spy.assert_called_with(
                CLIENT_SECRETS_INFO['web']['auth_uri'],
                access_type='offline',
                prompt='consent')
```